implementation/src/main/java/io/smallrye/health/checks/ThreadHealthCheck.java

```diff
@@ -0,0 +1,131 @@
+
+    }
+
+    private void addThreadCount(HealthCheckResponseBuilder responseBuilder, ThreadMXBean threadMXBean) {
+        try {
+            responseBuilder.withData(THREAD_COUNT, threadMXBean.getThreadCount());
+        } catch (Throwable t) {
+            responseBuilder.withData(THREAD_COUNT, UNAVAILABLE);
+        }
+    }
+
+    private void addPeakThreadCount(HealthCheckResponseBuilder responseBuilder, ThreadMXBean threadMXBean) {
+        try {
+            responseBuilder.withData(PEAK_THREAD_COUNT, threadMXBean.getPeakThreadCount());
+        } catch (Throwable t) {
+            responseBuilder.withData(PEAK_THREAD_COUNT, UNAVAILABLE);
+        }
+    }
+
+    private void addDaemonThreadCount(HealthCheckResponseBuilder responseBuilder, ThreadMXBean threadMXBean) {
+        try {
+            responseBuilder.withData(DAEMON_THREAD_COUNT, threadMXBean.getDaemonThreadCount());
+        } catch (Throwable t) {
+            responseBuilder.withData(DAEMON_THREAD_COUNT, UNAVAILABLE);
+        }
+    }
+
+    private void addTotalStartedThreadCount(HealthCheckResponseBuilder responseBuilder, ThreadMXBean threadMXBean) {
+        try {
+            responseBuilder.withData(STARTED_THREAD_COUNT, threadMXBean.getTotalStartedThreadCount());
+        } catch (Throwable t) {
+            responseBuilder.withData(STARTED_THREAD_COUNT, UNAVAILABLE);
+        }
+    }
+
+    private void addDeadlockedThreads(HealthCheckResponseBuilder responseBuilder, ThreadMXBean threadMXBean) {
+        try {
+            long[] deadlockedThreads = threadMXBean.findDeadlockedThreads();
+            responseBuilder.withData(DEADLOCKED_THREAD_COUNT, getNumberOfThreads(deadlockedThreads));
+        } catch (Throwable t) {
+            responseBuilder.withData(DEADLOCKED_THREAD_COUNT, UNAVAILABLE);
+        }
+    }
+
+    private void addMonitorDeadlockedThreads(HealthCheckResponseBuilder responseBuilder, ThreadMXBean threadMXBean) {
+        try {
+            long[] monitorDeadlockedThreads = threadMXBean.findMonitorDeadlockedThreads();
+            responseBuilder.withData(MONITOR_DEADLOCKED_THREAD_COUNT, getNumberOfThreads(monitorDeadlockedThreads));
+        } catch (Throwable t) {
+            responseBuilder.withData(MONITOR_DEADLOCKED_THREAD_COUNT, UNAVAILABLE);
+        }
+    }
+
+    private int getNumberOfThreads(long[] ids) {
+        if (ids == null)
+            return 0;
+        return ids.length;
+    }
+
+    private static final String THREAD_COUNT = "thread count";
+    private static final String PEAK_THREAD_COUNT = "peak thread count";
+    private static final String DAEMON_THREAD_COUNT = "daemon thread count";
+    private static final String STARTED_THREAD_COUNT = "started thread count";
+    private static final String DEADLOCKED_THREAD_COUNT = "deadlocked thread count";
+    private static final String MONITOR_DEADLOCKED_THREAD_COUNT = "monitor deadlocked thread count";
+    private static final String UNAVAILABLE = "Unavailable";
```

Comment:
I am used to put private constants after public constants at the start of the file but it's not a big deal.